# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: editor/libeditor/EditorBase.h
# Commit: cc2e8b7c548e
# Full Hash: cc2e8b7c548efb12f21236ea501751dd68773f28
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2022-09-22 09:56:47
# Regressor Bug: 1789967
# File Overlap Count: 1
# Description:
#   Bug 1789967 - part 1: Make `TextEditor` and `HTMLEditor` implement `EditorBase::InitEditorContentAndSelection` by themselves r=m_kato
#   
#   The method is enough simple, and uses bad cast from point of view of OOP.
#   Therefore, this patch make the sub classes implement the method only for each.
#   
# ==============================================================================

diff -r 2c56d379e755 -r cc2e8b7c548e editor/libeditor/EditorBase.h
--- a/editor/libeditor/EditorBase.h	Thu Sep 22 05:57:03 2022 +0000
+++ b/editor/libeditor/EditorBase.h	Thu Sep 22 06:06:54 2022 +0000
@@ -2245,13 +2245,6 @@
 
   [[nodiscard]] MOZ_CAN_RUN_SCRIPT nsresult EnsureEmptyTextFirstChild();
 
-  /**
-   * InitEditorContentAndSelection() may insert a padding `<br>` element for
-   * if it's required in the anonymous `<div>` element or `<body>` element and
-   * collapse selection at the end if there is no selection ranges.
-   */
-  [[nodiscard]] MOZ_CAN_RUN_SCRIPT nsresult InitEditorContentAndSelection();
-
   int32_t WrapWidth() const { return mWrapColumn; }
 
   /**